src/sentry/api/endpoints/project_group_index.py

```diff
@@ -328,7 +328,9 @@ def get(self, request, project):
             environment_id = self._get_environment_id_from_request(
                 request, project.organization_id)
         except Environment.DoesNotExist:
-            cursor_result = []
+            # XXX: The 1000 magic number is an abstraction leak from
+            # `sentry.api.paginator.BasePaginator.get_result`.
+            cursor_result = CursorResult([], None, None, 0, 1000)
```

Comment:
Maybe use the parameter name here so it's _slightly_ less magical.

```py
cursor_result = CursorResult([], None, None, 0, max_hits=1000)
```

Reply:
Good thinking, will improve

Reply:
dd32273